File: 4.8-bc/ts/stage-1-flat/src/repository/subscription.ts
import type { DbPool } from "../db.js";
import type { Subscription } from "../model/subscription.js";

export class SubscriptionRepository {
  constructor(private readonly db: DbPool) {}

  async create(s: Subscription): Promise<void> {
    await this.db.query(
      `INSERT INTO subscriptions (id, user_id, plan, next_charge_at, created_at)
       VALUES ($1, $2, $3, $4, $5)`,
      [s.id, s.userId, s.plan, s.nextChargeAt, s.createdAt],
    );
  }
}

File: 4.8-bc/py/stage-1-flat/app/repository/subscription.py
from __future__ import annotations

from sqlalchemy import text
from sqlalchemy.ext.asyncio import AsyncSession

from app.model.subscription import Subscription


class SubscriptionRepo:
    def __init__(self, session: AsyncSession) -> None:
        self._session = session

    async def create(self, s: Subscription) -> None:
        await self._session.execute(
            text(
                "INSERT INTO subscriptions "
                "(id, user_id, plan, next_charge_at, created_at) "
                "VALUES (:id, :user_id, :plan, :next_charge_at, :created_at)"
            ),
            {
                "id": s.id,
                "user_id": s.user_id,
                "plan": s.plan,
                "next_charge_at": s.next_charge_at,
                "created_at": s.created_at,
            },
        )
        await self._session.commit()
